bin/prettier.js

```diff
@@ -76,19 +79,55 @@ if (stdin) {
         output = format(input);
       } catch (e) {
         process.exitCode = 2;
-        console.error(e);
+        if (check) {
+          console.log(filename);
+        }
+        console.log(e);
         return;
       }
 
+      if (check) {
+        let output2;
+        try {
+          output2 = format(output);
+        } catch (e) {
+          function padLeft(nr, n, str){
+            return Array(n-String(nr).length+1).join(str||'0')+nr;
+          }
+          console.log(filename);
+          process.exitCode = 2;
+          const match = e.toString().match(/\(([0-9]+):/);
+          if (match) {
+            const line = +match[1];
```

Comment:
Is this getting the line number from a parser error message? At first glance it looks like it's trying to get the line number for the error object, and wouldn't that point to a place within `parser.js`, not the output?